### DISCUSSION TITLE
Octopus data coming in 1h late? Electricity mismatch with monitor plugs in energy dashboard

### DISCUSSION THREAD

**Discussion by Chaoscontrol** [BottlecapDave/HomeAssistant-OctopusEnergy#1109] [Q&A] [answered]
I've been noticing regular negative values in my energy dashboard. 

First thing I did was to check the actual sensor values, and there are no outliers to clean. You can see the electricity graph, and it's all positive. 

But within the untracked consumption, it shows negative. 

After thinking a bit, I could guess it's because some of my smart plugs are reporting a certain amount of energy at 11am. Then Octopus reports the spike at 12am. So the only way HA can resolve that is showing a negative value at 11am, and then leaving the spike at 12am "untracked", when in reality, it is tracked. It just seems to be offset. 

Is this the case? Is it normal? Is there anything I can do to fix it? 

This happens once or twice every day (noticeable in the graph), but in reality I guess it's happening along the entire graph. 

![image](https://github.com/user-attachments/assets/c638e976-7d61-46cf-8367-d613871f1d82)
![image](https://github.com/user-attachments/assets/93ba3d03-01f8-46fa-b926-5adb431aa34c)

**Comment by BottlecapDave** [comment]
Hello. The only thing that would cause this is if the combination of your tracked sensors is more than your meter (or mini if you have one) is reporting for the same period. 

There is an open [HA issue](https://github.com/home-assistant/core/issues/125362) around this by people who don't use this integration, so i think it's a HA issue

**Reply by Chaoscontrol** [reply]
That's what I mean. But it's not that the values are wrong. It's that they are delayed. And since the tracked sensors receive the data instantly via MQTT, I guess that the one coming in delayed is the Octopus (total) sensor. I do have a Mini.

**Reply by Chaoscontrol** [reply]
I've read that thread, but honestly I think that's a different issue. This is not a HA issue. HA is just trying to explain that difference. But looking at the total energy graph, you can see how the Octopus consumption is actually delayed (a certain amount, can't say how much, but close to 1h) from the individual devices.

**Reply by Chaoscontrol** [reply]
![image](https://github.com/user-attachments/assets/c7329516-e2c4-4913-9a39-92b996cedeeb)

See here, I offset it 1h and it clearly shows how tracked is always below the total now, and still keeping correlation.

**Reply by Chaoscontrol** [reply]
I went to check on Octopus app and it is even clearer here. 

The spike of 1kWh is at 11:00 + 11:30 sections, just like the tracked entities tracked. But the octopus sensor is actually attributing the spike to the 12:00 period. It feels like an integration or API issue. 

![image](https://github.com/user-attachments/assets/8ca83d65-ba9a-46a1-9196-b3ddab74727a)

**Reply by Chaoscontrol** [reply]
I just realised that I might be the culprit of this. A few months ago I disabled the automatic polling for the integration, and instead I enabled an automation to poll only once every hour, at XX:00. I did this to reduce db bloat I believe, though I don't remember much anymore. I'm not sure if that might be the issue of why the entire hour was attributed to the next hour and it caused the offset...

I have now enabled back the automation polling. Will wait 24h and check back again with the app and see the difference.

**Reply by Chaoscontrol** [reply, marked as answer]
Yep, I conclude it was my own mistake of disabling automatic polling, so when HA received the data after 1h it attributed it to the next hour. 

Today the graph matches perfectly what's in the Octopus app and so far there are no negative untracked portions to adjust the mismatches. 

![image](https://github.com/user-attachments/assets/330e6d3f-5cf3-4c1b-9b28-e50151d7747e)
